// speech.proto
syntax = "proto3";

package speech;

option go_package = "example.com/speechpb;speechpb";

// (float array) 
message AudioChunk {
  repeated float samples = 1;
  // timestamp can be used to align the audio chunks and testing
}

message Transcript {
  int32 start_time_millis = 1;
  int32 end_time_millis = 2;
  string text = 3;
}

// Bidirectional streaming
service SpeechToText {
  rpc StreamingRecognize(stream AudioChunk) returns (stream Transcript);
}

